### ISSUE TITLE
Exception after I restart 

### ISSUE THREAD

**Issue by tstee** [reicast/reicast-emulator#982]

**Comment by LoungeKatt** [comment]
It appears like the null pointer is the gamepad. When you go into settings, is there something being detected as the first gamepad? It sounds like it is automatically detecting another input method or the default input method is not being defined correctly.

**Comment by tstee** [comment]
Yeah... I configured the built-in gamepad as the first controller. It worked nicely first time around.

After a reboot, neither the games or the bios work anymore, it gives this exception when booting. Even removing the controller and assigning again or removing it at all and using the touchscreen. 

Strangely enough, if I remove from port A and assign the gamepad to port B it works. 

Even removing all assigned gamepad it crashes like that, so the only way to work is assigning the physycal gamepad to port 2.

**Comment by EPe9686518** [comment]
Same exact thing we are dealing with in this thread with GPD  https://github.com/reicast/reicast-emulator/issues/967

LoungeKatt the GPD uses Bitgames Center (Gamepad Assistant) to config the physical gamepads for the GPD handheld. You have 3 options, Playstation 3 controller, Xbox 360 or "Null". I have tested both the PS3 and the 360 with Reicast and the same error happens. Based on what you said, heres what I think might be happening. The GPD has a touch screen LCD display, and you can use the touch screen controls in Reicast as well as a gamepad. I think what is happening is Reicast is detecting the touch controls as well as the "physical" gamepad controls and this is causing the issue. Maybe if there was a way to disable touch controls in Reicast for android this might solve the problem, but this is just a guess. I do believe this is what is causing the issues here though, some conflict with the touch controls and the physical controls assigned by the gamepad software.

I can also confirm that removing the controller from port A and plugging it in to port B does work fine and prevents it from crashing. But some games wont work with the controller plugged into port B obviously.

**Comment by skelton** [comment]
Don't use bitgames assistant since it uses an ime that can provoke uses.  I use gpd xd and legacyrom with no bitgames assistant and reicast works perfectly. Uninstall bitgames assistant (it's justo a warez china App) and you will be ok.

**Comment by tstee** [comment]
Hi skelton,

After flashing the legacy rom I can confirm that the latest build works fine even after restarting the device.

Thanks for the help. I really appreciate it.

Any way to donate to you guys?

**Comment by EPe9686518** [comment]
I can also confirm on stock rom, reicast now works perfectly once Bitgame Gamepad assistant is removed on GPD XD. Thanks Skelton and to all that helped!
